components/brave_wallet/common/eth_sign_typed_data_helper.cc

```diff
@@ -194,12 +185,12 @@ EthSignTypedDataHelper::EncodeData(const std::string& primary_type_name,
 
 // Encode each field of a custom type, if a field is also a custom type it
 // will call EncodeData recursively until it reaches an atomic type
-std::optional<std::vector<uint8_t>> EthSignTypedDataHelper::EncodeField(
-    const std::string& type,
-    const base::Value& value) const {
+std::optional<EthSignTypedDataHelper::Eip712HashArray>
+EthSignTypedDataHelper::EncodeField(const std::string& type_string,
+                                    const base::Value& value) const {
```

Comment:
I think you could be taking `base::Value::List` here, and with that the check could be moved outside.

Reply:
This function should be split into many sub functions so it's clear what's going. Event clang-tidy complains
![image](https://github.com/user-attachments/assets/b6fcf483-0e03-4368-851b-1d4ac744dd42)

But as a follow up work

Reply:
Ah, understood.